### PR TITLE
[ch28785] flat attachments for FAM and PSEA

### PR DESCRIPTION
**Pull request by domdinicola**
[ch28785]

### CODE REVIEW

**Comment by domdinicola** [comment]
migrating existing attachments

```
from etools.applications.attachments.models import *
from etools.applications.attachments.utils import *
for country in Country.objects.exclude(schema_name='public'):
    set_country(country)
    for att in AttachmentFlat.objects.filter(source='Financial Assurance (FAM)'):
        att.agreement_reference_number = get_reference_number(att.attachment)
        att.save()
```

File: src/etools/applications/attachments/utils.py [hunk 1]
```diff
@@ -114,13 +116,17 @@ def get_agreement_obj(obj):
     )):
         if obj.content_object.intervention:
             return obj.content_object.intervention.agreement
+    elif isinstance(obj.content_object, Engagement):
```
Comment (emaciupe) [inline src/etools/applications/attachments/utils.py:119]: @domdinicola you could remove these lines and extend on line 108 e.g.
```
if isinstance(obj.content_object, (Agreement)):
       return obj.content_object
```
to
```
if isinstance(obj.content_object, (Agreement, Engagement, Assessment)):
       return obj.content_object
```

**Review by emaciupe — CHANGES_REQUESTED** [review]

File: src/etools/applications/attachments/utils.py (hunk 1 above)
Comment (domdinicola) [reply src/etools/applications/attachments/utils.py]: yep, I guess make sense [I was thinking to have one entry per module but anyways is not the case]

**Comment by domdinicola** [comment]
@emaciupe did you review also the script?

**Review by emaciupe — APPROVED** [review]
Looks good to me!

**Pull request merged**